compute-budget/src/compute_budget.rs

```diff
@@ -12,123 +12,177 @@ impl ::solana_frozen_abi::abi_example::AbiExample for ComputeBudget {
 
 /// Max instruction stack depth. This is the maximum nesting of instructions that can happen during
 /// a transaction.
-pub const MAX_INSTRUCTION_STACK_DEPTH: usize = 5;
+#[cfg_attr(feature = "svm-internal", qualifiers(pub))]
+const MAX_INSTRUCTION_STACK_DEPTH: usize = 5;
 
 /// Max call depth. This is the maximum nesting of SBF to SBF call that can happen within a program.
-pub const MAX_CALL_DEPTH: usize = 64;
+#[cfg_attr(feature = "svm-internal", qualifiers(pub))]
+const MAX_CALL_DEPTH: usize = 64;
 
 /// The size of one SBF stack frame.
-pub const STACK_FRAME_SIZE: usize = 4096;
+#[cfg_attr(feature = "svm-internal", qualifiers(pub))]
+const STACK_FRAME_SIZE: usize = 4096;
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(
+    any(feature = "dev-context-only-utils", feature = "svm-internal"),
+    field_qualifiers(
+        compute_unit_limit(pub),
+        log_64_units(pub),
+        create_program_address_units(pub),
+        invoke_units(pub),
+        max_instruction_stack_depth(pub),
+        max_instruction_trace_length(pub),
+        sha256_base_cost(pub),
+        sha256_byte_cost(pub),
+        sha256_max_slices(pub),
+        max_call_depth(pub),
+        stack_frame_size(pub),
+        log_pubkey_units(pub),
+        max_cpi_instruction_size(pub),
+        cpi_bytes_per_unit(pub),
+        sysvar_base_cost(pub),
+        secp256k1_recover_cost(pub),
+        syscall_base_cost(pub),
+        curve25519_edwards_validate_point_cost(pub),
+        curve25519_edwards_add_cost(pub),
+        curve25519_edwards_subtract_cost(pub),
+        curve25519_edwards_multiply_cost(pub),
+        curve25519_edwards_msm_base_cost(pub),
+        curve25519_edwards_msm_incremental_cost(pub),
+        curve25519_ristretto_validate_point_cost(pub),
+        curve25519_ristretto_add_cost(pub),
+        curve25519_ristretto_subtract_cost(pub),
+        curve25519_ristretto_multiply_cost(pub),
+        curve25519_ristretto_msm_base_cost(pub),
+        curve25519_ristretto_msm_incremental_cost(pub),
+        heap_size(pub),
+        heap_cost(pub),
+        mem_op_base_cost(pub),
+        alt_bn128_addition_cost(pub),
+        alt_bn128_multiplication_cost(pub),
+        alt_bn128_pairing_one_pair_cost_first(pub),
+        alt_bn128_pairing_one_pair_cost_other(pub),
+        big_modular_exponentiation_base_cost(pub),
+        big_modular_exponentiation_cost_divisor(pub),
+        poseidon_cost_coefficient_a(pub),
+        poseidon_cost_coefficient_c(pub),
+        get_remaining_compute_units_cost(pub),
+        alt_bn128_g1_compress(pub),
+        alt_bn128_g1_decompress(pub),
+        alt_bn128_g2_compress(pub),
+        alt_bn128_g2_decompress(pub),
+    )
+)]
+#[non_exhaustive]
 pub struct ComputeBudget {
```

Comment:
Could we leave these fields pub please? LiteSVM lets people overwrite the ComputeBudget fields.

I'm open to being told these fields are quite unstable and we should limit to a subset, since I doubt many users care about overwriting `poseidon_cost_coefficient_a` etc